compiler/z/codegen/OMRMemoryReference.cpp

```diff
@@ -578,16 +578,26 @@ OMR::Z::MemoryReference::MemoryReference(TR::Node * rootLoadOrStore, TR::CodeGen
          self()->createUnresolvedDataSnippet(rootLoadOrStore, cg, symRef, tempReg, isStore);
          }
 
+
+      // VSI vector load/store instructions, such as VSTRL, VLRL, and VSTEB, lack the I field and do not support
+      // index registers. As a result of this, we need to set separate index register so that an intermediate LA instruction
+      // is emitted to add up the index and base registers.
+      bool isPackedDecimalVectorLoadStore = rootLoadOrStore->getType().isAnyPacked()
+              && (TR::Compiler->target.cpu.getS390SupportsVectorPackedDecimalFacility() && !TR::Options::getCmdLineOptions()->getOption(TR_DisableVectorBCD))
+              && (rootLoadOrStore->getOpCode().isLoad() || rootLoadOrStore->getOpCode().isStore());
```

Comment:
```
rootLoadOrStore->getOpCode().isLoad() || rootLoadOrStore->getOpCode().isStore()
```

This is always true in this constructor, right? At least the code above this seems to assume so. Is it possible to fold this?

Reply:
It should be fine to remove the load/store op-code check. Let me test it without this constraint.